django/db/models/fields/related.py

```diff
@@ -613,6 +613,17 @@ def resolve_related_fields(self):
             )
             to_field = (self.remote_field.model._meta.pk if to_field_name is None
                         else self.remote_field.model._meta.get_field(to_field_name))
+            if to_field and to_field.model != self.remote_field.model._meta.concrete_model:
+                raise exceptions.FieldError(
+                    "ForeignKey '%s.%s' pointing to a non local remote field '%s.%s'." %
```

Comment:
This can happen also for `OneToOneField` so we shouldn't put `ForeignKey` in a message, maybe:

```python
exceptions.FieldError(
    "'%s.%s' refers to field '%s' which is not local to model "
    "'%s'." % (
        self.model._meta.label,
        self.name,
        to_field.name,
        self.remote_field.model._meta.concrete_model._meta.label,
    )
)
```